Lifecycle updates for dbt Cloud (#5517)
fishtown-analytics/docs.getdbt.com: website/docs/docs/dbt-versions/product-lifecycles.md

- dbt Core: The [open-source](https://github.com/dbt-labs/dbt-core) software that’s freely available 
- dbt Cloud: The managed [SaaS solution](https://www.getdbt.com/signup) built on top of dbt Core

All dbt features fall into a lifecycle category determined by their availablibilty in the following products:

### dbt Cloud

dbt Cloud features all fall into one of the following categories:

> - **Beta:** Beta features are still in development and are only available to select customers. To join a beta, there might be a signup form, or dbt Labs may contact specific customers about testing. Some can be activated by enabling [experimental features](/docs/dbt-versions/experimental-features) in your account. Beta features are incomplete and might not be entirely stable; they should be used at the customer’s risk, as breaking changes could occur. Beta features might not be fully documented, technical support is limited, and service level objectives (SLOs) might not be provided. Download the [Beta Features Terms and Conditions](/assets/beta-tc.pdf) for more details.

Reviewer: ```suggestion
- **Beta:** Beta features are still in development and are only available to select customers. To join a beta, there might be a signup form or dbt Labs may contact specific customers about testing. Some features can be activated by enabling [experimental features](/docs/dbt-versions/experimental-features) in your account. Beta features are incomplete and might not be entirely stable; they should be used at the customer’s risk, as breaking changes could occur. Beta features might not be fully documented, technical support is limited, and service level objectives (SLOs) might not be provided. Download the [Beta Features Terms and Conditions](/assets/beta-tc.pdf) for more details.
```

> - **Beta:** Beta features are still in development and are only available to select customers. To join a beta, there might be a signup form, or dbt Labs may contact specific customers about testing. Some can be activated by enabling [experimental features](/docs/dbt-versions/experimental-features) in your environment. Beta features are incomplete and might not be entirely stable; they should be used at the customer’s risk, as breaking changes could occur. Beta features might not be fully documented, technical support is limited, and service level objectives (SLOs) might not be provided. Download the [Beta Features Terms and Conditions](/assets/beta-tc.pdf) for more details.

Author: ```suggestion
- **Beta:** Beta features are still in development and are only available to select customers. To join a beta, there might be a signup form, or dbt Labs may contact specific customers about testing. Some can be activated by enabling [experimental features](/docs/dbt-versions/experimental-features) in your account. Beta features are incomplete and might not be entirely stable; they should be used at the customer’s risk, as breaking changes could occur. Beta features might not be fully documented, technical support is limited, and service level objectives (SLOs) might not be provided. Download the [Beta Features Terms and Conditions](/assets/beta-tc.pdf) for more details.
```

- **Preview:**  Preview features are stable and considered functionally ready for production deployments. Some planned additions and modifications to feature behaviors could occur before they become generally available. New functionality that is not backward compatible could also be introduced. Preview features include documentation, technical support, and service level objectives (SLOs). Features in preview are provided at no extra cost, although they might become paid features when they become generally available.

> - **Generally Available (GA):** Generally available features provide stable features introduced to all qualified dbt Cloud accounts. Service level agreements (SLAs) apply to GA features, including documentation and technical support. Certain GA feature availability is determined by the dbt version of the environment. To always receive the most recent GA features, ensure your dbt Cloud [environments](/docs/dbt-cloud-environments) are set to ["Keep on latest version"](/docs/dbt-versions/upgrade-dbt-version-in-cloud#keep-on-latest-version).

Reviewer: ```suggestion
- **Generally available (GA):** Generally available features provide stable features introduced to all qualified dbt Cloud accounts. Service level agreements (SLAs) apply to GA features, including documentation and technical support. Certain GA feature availability is determined by the dbt version of the environment. To always receive the latest GA features, ensure your dbt Cloud [environments](/docs/dbt-cloud-environments) are set to ["Keep on latest version"](/docs/dbt-versions/upgrade-dbt-version-in-cloud#keep-on-latest-version).
```

> - **Generally Available (GA):** Generally available features provide stable features introduced to all qualified production environments. Service level agreements (SLAs) apply to GA features, including documentation and technical support. Certain GA feature availability is determined by the dbt version of the environment. To always receive the most recent GA features, ensure your dbt Cloud [environments](/docs/dbt-cloud-environments) are set to ["Keep on latest version"](/docs/dbt-versions/upgrade-dbt-version-in-cloud#keep-on-latest-version).

Author: ```suggestion
- **Generally Available (GA):** Generally available features provide stable features introduced to all qualified dbt Cloud accounts. Service level agreements (SLAs) apply to GA features, including documentation and technical support. Certain GA feature availability is determined by the dbt version of the environment. To always receive the most recent GA features, ensure your dbt Cloud [environments](/docs/dbt-cloud-environments) are set to ["Keep on latest version"](/docs/dbt-versions/upgrade-dbt-version-in-cloud#keep-on-latest-version).
```

> - **Deprecated:** Features in this state are no longer being developed or enhanced by dbt Labs and will continue to function as-is until their removal date.

Reviewer: curious, should we say something about "no technical support / not supported" ?

> - **Deprecated:** Features in this state are no longer being developed or enhanced by dbt Labs and will continue to function as-is until their removal date.

Author: ```suggestion
- **Deprecated:** Features in this state are no longer being developed or enhanced by dbt Labs. They will continue functioning as-is, and their documentation will persist until their removal date. However, they are no longer subject to technical support. 
```
